design/Soft_finality_guarantee.md

```diff
@@ -0,0 +1,161 @@
+# The soft-finality guarantee 
+
+The decentralised POBI protocol links the rollups to the L1 block used as proof, which means that any L1 reorg
+results in an implicit L2 reorg. The finality of an L2 rollup is fully linked to the finality of an L1 block.
+
+During bootstrapping, we want a central sequencer to guarantee finality in around ~1 second and for the bridge to be decentralised 
+and resistant to reorgs.
+
+Note: See bridge and finality designs for more details
+
+
+## Problem and Requirements
+
+The central sequencer can only guarantee a consistent ordering of the transactions it receives from the Obscuro users.
+It can't offer guarantees over the finality of cross-chain transactions. E.g., deposits to the bridge.
+
+Given that deposits affect the balances of accounts, which can be used in Obscuro transactions, there is a dependence 
+between the ordering of the L1 messages and the result of executing L2 transactions.
+
+In the "fast finality" design, the "soft finality" of an L2 transaction is defined as the moment when the sequencer produces
+a Light Batch(LB) containing that transaction, and that LB is distributed to the network to respond to user requests.
+
+We can see, that there is a tension between the guarantees of "soft-finality" and the re-org resistance of the bridge.
+
+In this document we'll add some nuance to the soft-finality guarantee.
+
+The ideal outcome is that executing a transaction can never result in a different outcome, independent on the L1.
+
+A more realistic outcome is to relax the guarantee such that the tx execution result might be different only under certain
+factors unlikely to happen unless there is a re-org attack.
+
+
+## Outcomes of the design
+
+- In the context of fast finality and bridge designs, define the "Soft finality" guarantee.
+- Define the challenge conditions for the sequencer generating competing light batches.
+- Based on the above define the structure of a Light batch.
+
+## Assumption
+
+Messages from the L1 are processed as they are found in the L1 blocks, and synthetic transactions are generated based on them.
+These synthetic transactions are implicitly included in the light batches.
+
+
+## Solutions
+
+Since the problem is expressed in terms of "tension", we'll first try to explore the different options, and
+understand where they sit on the tradeoff spectrum.
+
+
+### Option 1 - Adding delays
+
+The straightforward solution is to wait a while before processing deposits so that these transactions are "final" on the L1.
+Eventually, the cost of mounting a re-org attack on the L1 becomes prohibitively high, ensuring security.
+
+There are two main disadvantages.
+
+The first is that users must wait before bridge messages are processed, which is not a great UX and it disables certain applications. 
+The second is that the solution is only as secure as the delay.
```

Comment:
Am I right to think it's completely insecure for the rollup period (e.g. six hours), then suddenly very secure (e.g. after six hours + 7 eth blocks)?